File: backend/utils/sample_formatting.py
from __future__ import annotations
from utils.sample_display_rules import WELL_DISPLAY_PREFIX


def format_well_display(well: str | None) -> str | None:
    """Форматирует скважину для отображения: «скв. №{номер}»."""
    if not well or not well.strip():
        return None
    return f"{WELL_DISPLAY_PREFIX}{well.strip()}"

File: frontend/src/entities/Sample/lib/sampleFormatting.ts
/** Префикс номера скважины в месте отбора. */
export const WELL_DISPLAY_PREFIX = 'скв. №';

/** Форматирует скважину для отображения: «скв. №{номер}». */
export function formatWellDisplay(well: string | null | undefined): string | null {
  const value = well?.trim();
  if (!value) {
    return null;
  }
  return `${WELL_DISPLAY_PREFIX}${value}`;
}
